solidity/test/TokenBridgeRouter.t.sol

```diff
@@ -0,0 +1,206 @@
+            address(this),
+            address(destinationManager),
+            address(0)
+        );
+
+        originTokenBridgeRouter.enrollRemoteRouter(
+            destinationDomain,
+            TypeCasts.addressToBytes32(address(destinationTokenBridgeRouter))
+        );
+        destinationTokenBridgeRouter.enrollRemoteRouter(
+            originDomain,
+            TypeCasts.addressToBytes32(address(originTokenBridgeRouter))
+        );
+
+        originTokenBridgeRouter.setTokenBridgeAdapter(
+            bridge,
+            address(bridgeAdapter)
+        );
+
+        destinationTokenBridgeRouter.setTokenBridgeAdapter(
+            bridge,
+            address(bridgeAdapter)
+        );
+
+        token.mint(address(this), amount);
+    }
+
+    function testSetTokenBridgeAdapter() public {
+        // Expect the TokenBridgeAdapterSet event.
+        // Expect topic0 & data to match
+        vm.expectEmit(true, false, false, true);
+        emit TokenBridgeAdapterSet(bridge, address(bridgeAdapter));
+
+        // Set the token bridge adapter
+        originTokenBridgeRouter.setTokenBridgeAdapter(
+            bridge,
+            address(bridgeAdapter)
+        );
+
+        // Expect the bridge adapter to have been set
+        assertEq(
+            originTokenBridgeRouter.tokenBridgeAdapters(bridge),
+            address(bridgeAdapter)
+        );
+    }
+
+    // ==== dispatchWithTokens ====
+
+    function testDispatchWithTokensRevertsWithUnkownBridgeAdapter() public {
+        vm.expectRevert("No adapter found for bridge");
+        originTokenBridgeRouter.dispatchWithTokens(
+            destinationDomain,
+            TypeCasts.addressToBytes32(address(recipient)),
+            messageBody,
+            address(token),
+            amount,
+            "BazBridge" // some unknown bridge name
+        );
+    }
+
+    function testDispatchWithTokensRevertsWithFailedTransferIn() public {
+        vm.expectRevert("ERC20: insufficient allowance");
+        originTokenBridgeRouter.dispatchWithTokens(
+            destinationDomain,
+            TypeCasts.addressToBytes32(address(recipient)),
+            messageBody,
+            address(token),
+            amount,
+            bridge
+        );
+    }
+
+    function testDispatchWithTokensTransfersTokensToAdapter() public {
+        vm.expectCall(
+            address(token),
+            abi.encodeWithSelector(
+                token.transferFrom.selector,
+                address(this),
+                address(bridgeAdapter),
+                amount
+            )
+        );
+        token.approve(address(originTokenBridgeRouter), amount);
+        originTokenBridgeRouter.dispatchWithTokens(
+            destinationDomain,
+            TypeCasts.addressToBytes32(address(recipient)),
+            messageBody,
+            address(token),
+            amount,
+            bridge
+        );
+    }
+
+    function testDispatchWithTokenTransfersMovesTokens() public {
```

Comment:
Is this test missing st?

Reply:
I think originally it was just testing that the origin side does not fail